apps/core/src/hooks/useProductAnalyticsConfig.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+import { useAppsBackend } from './useAppsBackend';
+
+type ProductAnalyticsConfigResponse = { mustProvideCookieConsent: boolean };
+
+export function useProductAnalyticsConfig() {
+    const { request } = useAppsBackend();
+    return useQuery({
+        queryKey: ['apps-backend', 'product-analytics-config'],
+        queryFn: () =>
```

Comment:
You could maybe persist this to localstorage because it changes so infrequently, and it'd let you optimize these flows a little.

Reply:
I'm going to punt on local storage persistence since there's a lot to get in here & this is already late, I did cache this for as long as the session remains open though. Will look into this next week as a follow-up